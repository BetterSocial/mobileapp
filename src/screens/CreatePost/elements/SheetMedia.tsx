import * as React from 'react';
import RBSheet from 'react-native-raw-bottom-sheet';
import {ActivityIndicator, StyleSheet, Text, TouchableOpacity, View} from 'react-native';

import MemoIcCamera from '../../../assets/icons/Ic_camera';
import MemoIcCreatePoll from '../../../assets/icons/ic_create_poll';
import MemoIcMedia from '../../../assets/icons/Ic_media';
import {fonts} from '../../../utils/fonts';
import {COLORS} from '../../../utils/theme';

export type SheetMediaProps = {
  refMedia: React.RefObject<RBSheet>;
  uploadFromMedia: () => void;
  takePhoto: () => void;
  createPoll: () => void;
  medias?: string[];
  isLoadingUploadingMedia?: boolean;
  isLoadingUploadingPhoto?: boolean;
};

export type ListProps = {
  icon: React.ReactNode;
  label: string;
  onPress: () => void;
  isLoading?: boolean;
};

const SheetMedia = ({
  refMedia,
  uploadFromMedia,
  takePhoto,
  createPoll,
  medias = [],
  isLoadingUploadingMedia = false,
  isLoadingUploadingPhoto = false
}: SheetMediaProps) => (
  <RBSheet
    ref={refMedia}
    closeOnDragDown={true}
    closeOnPressMask={true}
    customStyles={{
      container: styles.containerSheet,
      draggableIcon: styles.draggableIcon
    }}>
    <View style={styles.container}>
      {medias.length === 0 && (
        <List
          label="Add a poll"
          icon={<MemoIcCreatePoll width={16.67} height={16.67} />}
          onPress={createPoll}
        />
      )}
      <List
        label="Upload media from library"
        icon={<MemoIcMedia width={16.67} height={16.67} />}
        onPress={uploadFromMedia}
        isLoading={isLoadingUploadingMedia}
      />
      <List
        label="Take a photo"
        icon={<MemoIcCamera width={16.67} height={16.67} />}
        onPress={takePhoto}
        isLoading={isLoadingUploadingPhoto}
      />
    </View>
  </RBSheet>
);

const List = ({icon, label, onPress, isLoading = false}: ListProps) => (
  <TouchableOpacity style={styles.list} onPress={onPress}>
    {icon}
    <Text style={styles.labelList}>{label}</Text>
    {isLoading ? <ActivityIndicator size="small" color="#0000ff" style={styles.loader} /> : <></>}
  </TouchableOpacity>
);
export default SheetMedia;

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 20,
    paddingTop: 20,
    paddingBottom: 38
  },
  list: {
    flexDirection: 'row',
    paddingVertical: 17.67,
    paddingLeft: 21.67,
    backgroundColor: COLORS.lightgrey,
    borderRadius: 8,
    marginBottom: 12,
    alignItems: 'center'
  },
  labelList: {
    fontFamily: fonts.inter[500],
    fontSize: 14,
    color: COLORS.black,
    paddingLeft: 8,
    flex: 1
  },
  containerSheet: {
    borderTopRightRadius: 20,
    borderTopLeftRadius: 20
  },
  draggableIcon: {
    backgroundColor: COLORS.alto
  },
  loader: {
    marginRight: 10
  }
});
