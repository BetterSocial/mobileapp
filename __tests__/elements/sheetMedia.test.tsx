import * as React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import SheetMedia from '../../src/elements/Post/SheetMedia';
configure({adapter: new Adapter()});
// jest.useFakeTimers();
describe('component SheetMedia', () => {
  it('SheetMedia snapshot', () => {
    const myRef = {
      current: {
        open: () => {},
        close: () => {},
      },
    };
    const component = shallow(
      <SheetMedia
        refMedia={myRef}
        uploadFromMedia={() => {}}
        takePhoto={() => {}}
        createPoll={() => {}}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('SheetMedia renders correctly', () => {
    const myRef = {
      current: {
        open: () => {},
        close: () => {},
      },
    };

    const tree = renderer
      .create(
        <SheetMedia
          refMedia={myRef}
          uploadFromMedia={() => {}}
          takePhoto={() => {}}
          createPoll={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
