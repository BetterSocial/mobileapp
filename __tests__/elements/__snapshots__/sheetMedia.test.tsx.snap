// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component SheetMedia SheetMedia renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "portrait-upside-down",
      "landscape",
      "landscape-left",
      "landscape-right",
    ]
  }
  transparent={true}
  visible={false}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "#00000077",
            "flex": 1,
          },
          undefined,
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "opacity": 1,
        }
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "height": 0,
          "overflow": "hidden",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ccc",
                "borderRadius": 5,
                "height": 5,
                "margin": 10,
                "width": 35,
              },
              Object {
                "backgroundColor": "#E0E0E0",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "paddingBottom": 38,
            "paddingHorizontal": 20,
            "paddingTop": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 12,
              "opacity": 1,
              "paddingLeft": 21.67,
              "paddingVertical": 17.67,
            }
          }
        >
          <View />
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Inter-Medium",
                "fontSize": 14,
                "paddingLeft": 8,
              }
            }
          >
            Add a poll
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 12,
              "opacity": 1,
              "paddingLeft": 21.67,
              "paddingVertical": 17.67,
            }
          }
        >
          <View />
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Inter-Medium",
                "fontSize": 14,
                "paddingLeft": 8,
              }
            }
          >
            Upload media from library
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 12,
              "opacity": 1,
              "paddingLeft": 21.67,
              "paddingVertical": 17.67,
            }
          }
        >
          <View />
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Inter-Medium",
                "fontSize": 14,
                "paddingLeft": 8,
              }
            }
          >
            Take a photo
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`component SheetMedia SheetMedia snapshot 1`] = `
<RBSheet
  animationType="none"
  closeDuration={200}
  closeOnDragDown={true}
  closeOnPressBack={true}
  closeOnPressMask={true}
  customStyles={
    Object {
      "container": Object {
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
      },
      "draggableIcon": Object {
        "backgroundColor": "#E0E0E0",
      },
    }
  }
  dragFromTopOnly={false}
  height={260}
  keyboardAvoidingViewEnabled={false}
  minClosingHeight={0}
  onClose={null}
  onOpen={null}
  openDuration={300}
>
  <View
    style={
      Object {
        "paddingBottom": 38,
        "paddingHorizontal": 20,
        "paddingTop": 20,
      }
    }
  >
    <List
      icon={
        <Memo(IcCreatePoll)
          height={16.67}
          width={16.67}
        />
      }
      label="Add a poll"
      onPress={[Function]}
    />
    <List
      icon={
        <Memo(Ic_media)
          height={16.67}
          width={16.67}
        />
      }
      label="Upload media from library"
      onPress={[Function]}
    />
    <List
      icon={
        <Memo(Ic_camera)
          height={16.67}
          width={16.67}
        />
      }
      label="Take a photo"
      onPress={[Function]}
    />
  </View>
</RBSheet>
`;
